In [1]:
# IMporting the libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
#Read Dataset  
data=pd.read_csv('Train.csv')
print(data.head())

           Tour_ID                   country age_group  \
0  tour_id1hffseyw                     ITALY     45-64   
1  tour_idnacd7zag            UNITED KINGDOM     25-44   
2  tour_id62vz7e71  UNITED STATES OF AMERICA       65+   
3  tour_idrc76tzix                    RWANDA     25-44   
4  tour_idn723m0n9  UNITED STATES OF AMERICA     45-64   

                travel_with  total_female  total_male  \
0             With Children           0.0         2.0   
1               With Spouse           1.0         1.0   
2               With Spouse           1.0         1.0   
3  With Spouse and Children           3.0         1.0   
4                     Alone           0.0         1.0   

                          purpose     main_activity  \
0  Visiting Friends and Relatives     Beach Tourism   
1            Leisure and Holidays  Wildlife Tourism   
2            Leisure and Holidays   Widlife Tourism   
3            Leisure and Holidays     Beach Tourism   
4            Leisure and Holidays

In [3]:
print(data.shape)

(18506, 21)


In [4]:
print(data.columns)

Index(['Tour_ID', 'country', 'age_group', 'travel_with', 'total_female',
       'total_male', 'purpose', 'main_activity', 'info_source',
       'tour_arrangement', 'package_transport_int', 'package_accomodation',
       'package_food', 'package_transport_tz', 'package_sightseeing',
       'package_guided_tour', 'package_insurance', 'night_mainland',
       'night_zanzibar', 'first_trip_tz', 'cost_category'],
      dtype='object')


In [5]:
data.dtypes

Tour_ID                   object
country                   object
age_group                 object
travel_with               object
total_female             float64
total_male               float64
purpose                   object
main_activity             object
info_source               object
tour_arrangement          object
package_transport_int     object
package_accomodation      object
package_food              object
package_transport_tz      object
package_sightseeing       object
package_guided_tour       object
package_insurance         object
night_mainland             int64
night_zanzibar             int64
first_trip_tz             object
cost_category             object
dtype: object

In [ ]:
l2=[]
l1=data['age_group'].tolist()
for i in l1:
    if i=='18-24':
        age='young'
    elif i=='25-44':
        age='adult'
    elif i=='45-64':
        age='senior'
    else:
        age='super_senior'
    l2.append(age)
data['age']=l2
data

# PreProcessing

##  1.Missing value Handling

In [6]:
data.isna().sum()

Tour_ID                     0
country                     0
age_group                   0
travel_with              1075
total_female                2
total_male                  6
purpose                     0
main_activity               0
info_source                 0
tour_arrangement            0
package_transport_int       0
package_accomodation        0
package_food                0
package_transport_tz        0
package_sightseeing         0
package_guided_tour         0
package_insurance           0
night_mainland              0
night_zanzibar              0
first_trip_tz               0
cost_category               0
dtype: int64

There are 1075 missing values in the column travel_with , 2 in the column total_female and 6 in the total_male

In [7]:
data['total_female']=data['total_female'].fillna(data['total_female'].median())

In [8]:
data['total_male']=data['total_male'].fillna(data['total_male'].median())

In [9]:
data.ffill(axis=0,inplace=True)

In [10]:
data.isna().sum()

Tour_ID                  0
country                  0
age_group                0
travel_with              0
total_female             0
total_male               0
purpose                  0
main_activity            0
info_source              0
tour_arrangement         0
package_transport_int    0
package_accomodation     0
package_food             0
package_transport_tz     0
package_sightseeing      0
package_guided_tour      0
package_insurance        0
night_mainland           0
night_zanzibar           0
first_trip_tz            0
cost_category            0
dtype: int64

Now all the missing values are replaced

## 2.Outlier detection and handling

In [ ]:
plt.boxplot

## 2.Encoding

In [ ]:
data['travel_with'].value_counts()

In [ ]:
# Label Encoding 'travel_with'
from sklearn.preprocessing import LabelEncoder
label_en=LabelEncoder()
a=['travel_with']
for i in np.arange(len(a)):
    data[a[i]]=label_en.fit_transform(data[a[i]])


In [ ]:
data['travel_with'].value_counts()

In [ ]:
data['purpose'].value_counts()

In [ ]:
# Label Encoding 'purpose'
label_en=LabelEncoder()
a=['purpose']
for i in np.arange(len(a)):
    data[a[i]]=label_en.fit_transform(data[a[i]])

In [ ]:
data['purpose'].value_counts()

In [ ]:
data['main_activity'].value_counts()

In [ ]:
# Label Encoding 'main_activity'
label_en=LabelEncoder()
a=['main_activity']
for i in np.arange(len(a)):
    data[a[i]]=label_en.fit_transform(data[a[i]])

In [ ]:
data['main_activity'].value_counts()

In [ ]:
data['info_source'].value_counts()

In [ ]:
# Label Encoding 'info_source'
label_en=LabelEncoder()
a=['info_source']
for i in np.arange(len(a)):
    data[a[i]]=label_en.fit_transform(data[a[i]])

In [ ]:
data['info_source'].value_counts()

In [ ]:
data['cost_category'].value_counts()

In [ ]:
# Label Encoding 'cost_category'
label_en=LabelEncoder()
a=['cost_category']
for i in np.arange(len(a)):
    data[a[i]]=label_en.fit_transform(data[a[i]])


In [ ]:
data['cost_category'].value_counts()

In [ ]:
data['age'].value_counts()

In [ ]:
label_en=LabelEncoder()
a=['age']
for i in np.arange(len(a)):
    data[a[i]]=label_en.fit_transform(data[a[i]])


In [ ]:
data['age'].value_counts()

In [ ]:
data['tour_arrangement'].value_counts()

In [ ]:
label_en=LabelEncoder()
a=['tour_arrangement']
for i in np.arange(len(a)):
    data[a[i]]=label_en.fit_transform(data[a[i]])


In [ ]:
data['tour_arrangement'].value_counts()

In [ ]:
# One hot encoding
one_hot_encoded_data = pd.get_dummies(data, columns = ['package_transport_int','package_accomodation','package_food','package_transport_tz', 'package_sightseeing',
       'package_guided_tour', 'package_insurance','first_trip_tz'])
one_hot_encoded_data

In [ ]:
data.columns

## 3.Feature reduction

In [ ]:
# Checking columns
data.columns

In [ ]:
# Droping 'age_group'
data.drop(['age_group'],axis=1,inplace=True)

In [ ]:
# Combining 'total_male'  and 'total_female'
data['total_size']=data['total_male']+data['total_female']
data.drop(['total_male'],axis=1,inplace=True)
data.drop(['total_female'],axis=1,inplace=True)

In [ ]:
data

In [ ]:
data.drop(['Tour_ID'],axis=1,inplace=True)

In [ ]:
data

In [ ]:

from sklearn.model_selection import train_test_split
corrmatrix=data.corr()
plt.subplots(figsize=(20,8))
sns.heatmap(corrmatrix,vmin=-0.4,vmax=0.9,annot=True,linewidth=0.2,cmap='YlGnBu')